src/modules/AutoinstConfig.rb

```diff
@@ -214,114 +219,157 @@ def setProposalList(l)
+              Ops.greater_than(
+                Ops.subtract(Ops.subtract(endComment, startComment), 1),
+                0
               )
-              UI.CloseDialog
-            elsif Builtins.size(slpData) == 1
-              cmdLine = Builtins.substring(
-                Ops.get_string(slpData, [0, "srvurl"], ""),
-                17
+              comment = Builtins.substring(
+                comment,
+                Ops.add(startComment, 1),
+                Ops.subtract(Ops.subtract(endComment, startComment), 1)
               )
             else
-              cmdLine = "slp query for 'autoyast' failed"
+              comment = ""
+            end
+
+            if Ops.less_than(Builtins.size(comment), 1)
+              comment = Builtins.sformat(
+                "bad description in SLP for %1",
+                url
+              )
             end
+
+            dummy = Builtins.add(dummy, Item(comment, false))
+            Ops.set(comment2url, comment, url)
+          else
+            url = Builtins.substring(Ops.get_string(m, "srvurl", ""), 17)
+            dummy = Builtins.add(dummy, Item(url, false))
+            Ops.set(comment2url, url, url)
           end
-          result = URL.Parse(cmdLine)
-          @OriginalURI = cmdLine
         end
+
+        dlg = Left(ComboBox(Id(:choose), _("Choose Profile"), dummy))
+
+        UI.OpenDialog(VBox(dlg, PushButton(Id(:ok), Label.OKButton)))
+        UI.UserInput
+
+        profile_location = Ops.get(
+          comment2url,
+          Convert.to_string(UI.QueryWidget(Id(:choose), :Value)),
+          ""
+        )
+
+        UI.CloseDialog
+
+      # just one provider
+      elsif Builtins.size(slpData) == 1
+        profile_location = Builtins.substring(
+          Ops.get_string(slpData, [0, "srvurl"], ""),
+          17
+        )
+
+      # Nothing returned by SLP query
+      else
+        log.error "slp query for 'autoyast' failed"
+        Report.Error(_("No 'autoyast' provider has been found via SLP."))
       end
 
+      profile_location
+    end
+
+    # Updates or extends the profile location according to defaults
+    # @param [String] AutoYast profile location as defined on commandline
+    # @return [String] updated profile location
+    def update_profile_location(profile_location)
+      profile_location = deep_copy(profile_location)
 
-      if Ops.get_string(result, "scheme", "") == ""
+      if profile_location.nil? || profile_location == ""
+        # FIXME: reevaluate this statement
+        #
         # Autoinstall mode was not activated from command line.
         # There must be a floppy with an 'autoinst.xml' in order
         # to be able to reach this point, so we set floppy with
         # autoinst.xml as the control file.
-
-        result = Builtins.add(result, "scheme", "floppy")
-        result = Builtins.add(result, "path", "/autoinst.xml")
+        profile_location = "floppy:///#{DEFAULT_PROFILE_NAME}"
```

Comment:
Although it should not be dangerous, I would not overwrite the argument I've received. Also having a lot of references to the same variable is not readable at all (well, that one is just a personal opinion).